### PR TITLE
Export LogLine type synonym

### PR DESCRIPTION
**Pull request by mbwgh**
If the symbol is not exported, you have to look at the source to see what you are working with.

The same applies to `DList`, which is visible from `unWriterLoggingT`, but this would clash with the `dlist` package.

### CODE REVIEW

File: Control/Monad/Logger.hs [hunk 1]
```diff
@@ -34,7 +34,7 @@ module Control.Monad.Logger
       MonadLogger(..)
     , MonadLoggerIO (..)
     , LogLevel(..)
-    , LogSource
+    , LogSource(..)
```
Comment (snoyberg) [inline Control/Monad/Logger.hs]: What's the purpose of this change?

File: Control/Monad/Logger.hs [hunk 2]
```diff
@@ -49,6 +49,7 @@ module Control.Monad.Logger
     , filterLogger
     , NoLoggingT (..)
     , mapNoLoggingT
+    , LogLine(..)
```
Comment (snoyberg) [inline Control/Monad/Logger.hs]: This is a type alias, it should be exported without the `(..)`

File: Control/Monad/Logger.hs [hunk 3]
```diff
@@ -883,8 +884,8 @@ logCS cs src lvl msg =
   monadLoggerLog (locFromCS cs) src lvl msg
 
 -- | Logs a message with location given by 'CallStack'.
--- See 'Control.Monad.Logger.CallStack' for more convenient
--- functions for 'CallStack' based logging.
+-- See "Control.Monad.Logger.CallStack" for more convenient
+-- functions for @CallStack@ based logging.
```
Comment (snoyberg) [inline Control/Monad/Logger.hs]: What's the purpose of this change?

File: Control/Monad/Logger.hs (hunk 1 above)
Comment (mbwgh) [reply Control/Monad/Logger.hs]: Sorry, I'm new. I didn't know that pushing to my fork would add this commit to this PR. The reason why I did this was that in my local haddocks, there was

```haskell
type LogSource = Text
```

where the `Text` part wasn't hyperlinked. This is what confused me into adding the `(..)`, even though I couldn't find anything about this in the haddock documentation or the Haskell Report. But now I think that simply recompilation fixed it, and I can't reproduce the non-hyperlinked `Text` either. So all in all, this is just plain wrong.

File: Control/Monad/Logger.hs (hunk 2 above)
Comment (mbwgh) [reply Control/Monad/Logger.hs]: Indeed, and I merely wrote it like this because I did so for `LogSource` as well, under the assumption that it "couldn't hurt".

File: Control/Monad/Logger.hs (hunk 3 above)
Comment (mbwgh) [reply Control/Monad/Logger.hs]: This haddock currently tries to quote the `CallStack` module, but the link just leads back to `Control.Monad.Logger`. According to the haddock [documentation](https://haskell-haddock.readthedocs.io/en/latest/markup.html#linking-to-modules) you should quote modules with double quotes.

Comment (snoyberg) [reply Control/Monad/Logger.hs]: That's fine on line 887. I'm asking about line 888: why did you change it from `'CallStack'` to `@CallStack@`?

Comment (mbwgh) [reply Control/Monad/Logger.hs]: That isn't really a strict "improvement" of course. I merely thought it looked awkward that `CallStack` is quoted three times in a row in a single haddock, the third quotation being redundant because of the first one.

This really should've gone into a more general "doc improvements" patch. For instance the haddocks for `runWriterLoggingT` and `execWriterLoggingT` haddocks have issues as well.
I spent some time yesterday trying to figure out if this review section would become messed up if I ammended/rebased the tip of my fork. I now believe I should do so, and do everything not related to exporting the `LogLine` synonym in a feature-branch. I just wanted to get some confirmation first if that's how it should be done.

Sorry for the beginner mistakes.

Comment (snoyberg) [reply Control/Monad/Logger.hs]: Yes, rebasing would be a good idea.

**Comment by dzhus** [comment]
@mbwgh Could you rebase again please so that it could be merged maybe? 0.3.30 is now taken apparently :)
